### PR TITLE
feat(rspack): dev & build

### PR DESCRIPTION
**Pull request by underfin**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

- use [vue-rsloader](https://github.com/skywalker512/vue-rsloader) as temporary workaround for vue-loader.
- bump rspack version to let hmr works.
- tracking issue with rspack, at https://github.com/web-infra-dev/rspack/issues/2592

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/rspack/src/plugins/chunk.ts [hunk 1]
```diff
@@ -3,26 +3,26 @@ import webpack from '@rspack/core'
...
+// };`
 
 export class ChunkErrorPlugin {
   apply (compiler: Compiler) {
     compiler.hooks.thisCompilation.tap(pluginName, compilation =>
-      compilation.mainTemplate.hooks.localVars.tap(
-        { name: pluginName, stage: 1 },
-        source => source + script
-      )
+      // compilation.mainTemplate.hooks.localVars.tap(
```
Comment (underfin) [inline packages/rspack/src/plugins/chunk.ts:22]: Look like `compilation.mainTemplate.hooks.localVars` is update `webpack.RuntimeGlobals.ensureChunk` runtime module, but the implement may inject many `script` logic code. Look like use `runtime_requirements_in_tree` hooks to add the update runtime module.

File: packages/rspack/src/presets/style.ts [hunk 2]
```diff
@@ -72,25 +72,26 @@ function createdStyleRule (lang: string, test: RegExp, processorLoader: any, ctx
     options.webpack.loaders.cssModules.importLoaders =
     styleLoaders.length
 
-  const cssLoaders = createCssLoadersRule(ctx, options.webpack.loaders.css)
-  const cssModuleLoaders = createCssLoadersRule(ctx, options.webpack.loaders.cssModules)
+  // const cssLoaders = createCssLoadersRule(ctx, options.webpack.loaders.css)
+  // const cssModuleLoaders = createCssLoadersRule(ctx, options.webpack.loaders.cssModules)
```
Comment (underfin) [inline packages/rspack/src/presets/style.ts:76]: Rspack support `MiniCssExtractPlugin` + `CssMinimizerPlugin` + `CssModules` by default.

File: packages/rspack/src/presets/vue.ts [hunk 3]
```diff
@@ -8,15 +8,15 @@ export function vue (ctx: RspackConfigContext) {
...
+  config.plugins.push(new (VueLoaderPlugin.default || VueLoaderPlugin)())
 
   config.module!.rules!.push({
     test: /\.vue$/i,
-    use: ['/Users/daniel/code/nuxt.js/packages/rspack/src/loaders/vue.cjs']
-    // options: {
-    //   reactivityTransform: ctx.nuxt.options.experimental.reactivityTransform,
-    //   ...options.webpack.loaders.vue
-    // }
+    use: require.resolve('vue-rsloader'),
```
Comment (underfin) [inline packages/rspack/src/presets/vue.ts:15]: Here use `vue-rsloader` as a workaround for `vue-loader`, we will support `vue-loader` at furtrue

File: packages/rspack/src/rspack.ts [hunk 4]
```diff
@@ -36,6 +37,7 @@ export async function bundle (nuxt: Nuxt) {
     return getRspackConfig(ctx)
   })
 
+  // @ts-ignore
   await nuxt.callHook('rspack:config', webpackConfigs)
```
Comment (underfin) [inline packages/rspack/src/rspack.ts:41]: Here miss some hook key for rspack.

File: packages/rspack/src/rspack.ts [hunk 5]
```diff
@@ -64,7 +66,7 @@ export async function bundle (nuxt: Nuxt) {
 
     // In dev, write files in memory FS
     if (nuxt.options.dev) {
-      compiler.outputFileSystem = mfs as unknown as OutputFileSystem
+      compiler.outputFileSystem = mfs as any /* as OutputFileSystem */
```
Comment (underfin) [inline packages/rspack/src/rspack.ts:69]: I'm not sure it need at rspack, `getRspackMemoryAssets` should work for dev server visit resource. Maybe it used for online edit?

File: packages/rspack/src/rspack.ts [hunk 6]
```diff
@@ -103,20 +106,22 @@ async function createDevMiddleware (compiler: Compiler) {
...
+  // @ts-ignore
+  const hotMiddleware = webpackHotMiddleware(compiler, {
+    log: false,
+    heartbeat: 10000,
+    path: joinURL(nuxt.options.app.baseURL, '__webpack_hmr', compiler.options.name!),
+    ...hotMiddlewareOptions
+  })
 
   // Register devMiddleware on server
+  // @ts-ignore
```
Comment (underfin) [inline packages/rspack/src/rspack.ts:119]: The `webpackHotMiddleware` is compatiable with rspack.

File: packages/rspack/src/presets/node.ts [hunk 7]
```diff
@@ -4,34 +4,34 @@ export function node (ctx: RspackConfigContext) {
...
-      const: true,
-      destructuring: true,
-      dynamicImport: true,
-      forOf: true
-    },
-    library: {
-      type: 'module'
-    }
-  }
+  // config.output = {
```
[the remark this answers is not in the document]
Comment (underfin) [reply packages/rspack/src/presets/node.ts]: `module` output already supported

**Comment by HigherOrderLogic** [comment]
Any update on this?

**Pull request closed without merging**